client/src/main/scala/org/http4s/client/oauth1/ProtocolParameter.scala

```diff
@@ -44,7 +45,12 @@ object ProtocolParameter {
 
   case class Custom(headerName: String, headerValue: String) extends ProtocolParameter
 
-  case class SignatureMethod(override val headerValue: String = "HMAC-SHA1")
+  object SignatureMethod {
+    val SHA1: SignatureMethod = SignatureMethod()
+    val SHA256: SignatureMethod = SignatureMethod(headerValue = "HMAC-SHA256", algorithm = "HmacSHA256")
+  }
+
+  case class SignatureMethod(override val headerValue: String = "HMAC-SHA1", algorithm: String = "HmacSHA1")
```

Comment:
This one here is the binary incompatibility, right?  Case classes are such a pain for maintaining compat.

Supporting arbitrary new methods is tough, but we can support some common ones now.

```scala
private val methodsToAlgorithms = Map(
  HmacSha1 -> "HmacSHA1",
  HmacSha256 -> "HmacSHA256",
  // I think RsaSha1 and Plaintext are the other common ones?
  // We can add them later, so don't let it distract you from the
  // one that you need.
)

case class SignatureMethod(val headerValue: String) extends ProtocolParameter {
  override val headerName: String = "oauth_signature_method"
  def algorithm: Option[String] = methodsToAlgorithms(headerValue)
}
```

Then if `algorithm` is `None`, we can raise an error in `signRequest`, or else map our way to victory.  Or we could just default algorithm to `HmacSHA1`, which is what it's hardcoded to today.  I'm comfortable raising the error, because if people go to the trouble of changing the method, they need a different algorithm.

That strategy is not as good as what you've done, but I think it's a path to compatibility.

Reply:
As for 1.0 design, maybe `algorithm` as a `String` isn't right: what if it's plaintext?   Maybe the method should have some sort of `Request[F] => F[Request[F]]` shape?

Keeping that constructor public is extensible, but may allow nonsense combinations.

Reply:
I implemented all the HMAC-SHA* algorithms.  The PLAINTEXT and Rsa* can be an adventure for someone else :)